### PR TITLE
Remove, hide or unstablize deprecated API

### PR DESCRIPTION
**Pull request by trustin**
- Removed all deprecated API if possible.
- Hid all deprecated API if possible.
- Some notable changes worth mentioning:
  - `DefaultHttp{Request,Response}` has been moved to `internal.common`.
  - Renamed `DefaultRpcResponse` to `CompletableRpcResponse` and left only
    the default constructor.
  - Removed `http` from `RequestContextExporter` and its related classes,
    e.g. `httpRequestHeaders` -> `requestHeaders`
  - Fixed bugs where `RequestContextAware{Future,Promise,ProgressivePromise}`
    does not return itself but its delegate.
    - `removeListener(s)` now throws `UnsupportedOperationException`,
      because there's no way to remove the wrapped listener by reference.
  - Moved `RequestContext.makeContextAware(GenericFutureListener)` to
    `RequestContextAwareFutureListener` and `RequestContextAwareProgressiveFutureListener`.
  - Fixed a bug where `RequestContextExporter.attributes()` throws an exception or a map with incorrect aliases.

### CODE REVIEW

**Comment by trustin** [comment]
Note that I did not remove:
- `*RequestContext.set*Timeout*()` because we might undeprecate them soon.
- `MoreNamingConventions` to give users more time to migrate.

**Comment by ikhoon** [comment]
Still LGTM :-)

File: core/src/main/java/com/linecorp/armeria/common/logging/RequestContextExporterBuilder.java [hunk 1]
```diff
@@ -316,7 +221,7 @@ public RequestContextExporterBuilder addKeyPattern(String keyPattern) {
         }
 
         if (exportKey == null) {
-            exportKey = components[0].substring(PREFIX_ATTRS.length());
+            exportKey = components[0];
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/common/logging/RequestContextExporterBuilder.java]: 👍

**Comment by trustin** [comment]
Thanks for reviewing!

**Pull request merged**